### PR TITLE
Clarify use of TrainingArguments.disable_tqdm in Jupyter Notebooks  

### LINKED ISSUE #8831: logging.set_verbosity_error() displays dict instead of NotebookTrainingTracker
## Environment info

     
- `transformers` version: 4.0.0-rc-1
- Platform: Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.6.9
- PyTorch version (GPU?): 1.7.0+cu101 (True)
- Tensorflow version (GPU?): 2.3.0 (True)
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help

@sgugger 

## Information

Model I am using (Bert, XLNet ...): `distilbert-base-uncased`

The problem arises when using:
* [] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

Following the [docs](https://huggingface.co/transformers/main_classes/logging.html#logging) I was looking for a way to turn off the warnings that `transformers` shows when loading a new model and believe that `logging.set_verbosity_error()` should do the trick. 

However, when working in a _Jupyter notebook environment_, I find that setting the logging level to error produces unexpected output from the `Trainer`, namely that I get a `dict` like

```
{'loss': 0.33437405395507813, 'learning_rate': 1.308411214953271e-06, 'epoch': 0.9345794392523364}
{'eval_loss': 0.509843111038208, 'eval_matthews_correlation': 0.5011235129840701, 'epoch': 1.0}
{'epoch': 1.0}
```

instead of the progress bar and table of metrics:

Screen Shot 2020-11-28 at 4 21 34 pm

I encountered the problem in my own experiments, but have also been able to reproduce it in @sgugger's tutorial on the GLUE tasks: https://colab.research.google.com/github/huggingface/notebooks/blob/master/examples/text_classification.ipynb

The tasks I am working on is:
* [x] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

The task is GLUE

## To reproduce

1. Set the logging verbosity to _error_ in the first cell of the notebook, i.e. with

```
# Turn off warnings
import transformers
transformers.logging.set_verbosity_error()
```

2. Load and encode dataset
3. Configure trainer
4. Run training
```
# With logging.set_verbosity_error() we lose the metrics table :(
trainer.train()
# Output
{'loss': 0.33437405395507813, 'learning_rate': 1.308411214953271e-06, 'epoch': 0.9345794392523364}
{'eval_loss': 0.509843111038208, 'eval_matthews_correlation': 0.5011235129840701, 'epoch': 1.0}
{'epoch': 1.0}
TrainOutput(global_step=535, training_loss=0.34615044994889016)
```

I have trimmed down @sgugger's tutorial to create a reproducible example: https://colab.research.google.com/gist/lewtun/21d44a20f94f480dfa2891f587323ffd/logging-bug-in-trainer.ipynb

## Expected behavior

Changing the logging level should not interfere with the display of the progress bar or table of metrics in Jupyter notebooks.

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

Closes #8831 and adds some minor tweaks / improvements to the `TrainingArguments` classes.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

@sgugger

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -51,9 +51,9 @@ class TrainingArguments:
     TrainingArguments is the subset of the arguments we use in our example scripts **which relate to the training loop
     itself**.
 
-    Using :class:`~transformers.HfArgumentParser` we can turn this class into
-    `argparse <https://docs.python.org/3/library/argparse.html#module-argparse>`_ arguments that can be specified on
-    the command line.
+    Using :class:`~transformers.HfArgumentParser` we can turn this class into `argparse
+    <https://docs.python.org/3/library/argparse.html#module-argparse>`_ arguments that can be specified on the command
```
Comment (sgugger) [inline src/transformers/training_args.py:55]:
```suggestion
    <https://docs.python.org/3/library/argparse.html#module-argparse>`__ arguments that can be specified on the command
```

File: src/transformers/training_args.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ class TrainingArguments:
         warmup_steps (:obj:`int`, `optional`, defaults to 0):
             Number of steps used for a linear warmup from 0 to :obj:`learning_rate`.
         logging_dir (:obj:`str`, `optional`):
-            `TensorBoard <https://www.tensorflow.org/tensorboard>`_ log directory. Will default to 
+            `TensorBoard <https://www.tensorflow.org/tensorboard>`_ log directory. Will default to
```
Comment (sgugger) [inline src/transformers/training_args.py:128]:
```suggestion
            `TensorBoard <https://www.tensorflow.org/tensorboard>`__ log directory. Will default to
```

File: src/transformers/training_args.py [hunk 3]
```diff
@@ -172,8 +172,8 @@ class TrainingArguments:
             :class:`~transformers.notebook.NotebookTrainingTracker` in Jupyter Notebooks. Will default to :obj:`True`
             if the logging level is set to warn or lower (default), :obj:`False` otherwise.
         remove_unused_columns (:obj:`bool`, `optional`, defaults to :obj:`True`):
-            If using `datasets.Dataset` datasets, whether or not to automatically remove the columns unused by the model
-            forward method.
+            If using `datasets.Dataset` datasets, whether or not to automatically remove the columns unused by the
```
Comment (sgugger) [inline src/transformers/training_args.py:175]:
```suggestion
            If using :obj:`datasets.Dataset` datasets, whether or not to automatically remove the columns unused by the
```

File: src/transformers/training_args_tf.py [hunk 4]
```diff
@@ -33,9 +33,9 @@ class TFTrainingArguments(TrainingArguments):
     TrainingArguments is the subset of the arguments we use in our example scripts **which relate to the training loop
     itself**.
 
-    Using :class:`~transformers.HfArgumentParser` we can turn this class into
-    `argparse <https://docs.python.org/3/library/argparse.html#module-argparse>`_ arguments that can be specified on
-    the command line.
+    Using :class:`~transformers.HfArgumentParser` we can turn this class into `argparse
+    <https://docs.python.org/3/library/argparse.html#module-argparse>`_ arguments that can be specified on the command
```
Comment (sgugger) [inline src/transformers/training_args_tf.py:37]:
```suggestion
    <https://docs.python.org/3/library/argparse.html#module-argparse>`__ arguments that can be specified on the command
```

File: src/transformers/training_args_tf.py [hunk 5]
```diff
@@ -97,7 +97,7 @@ class TFTrainingArguments(TrainingArguments):
         warmup_steps (:obj:`int`, `optional`, defaults to 0):
             Number of steps used for a linear warmup from 0 to :obj:`learning_rate`.
         logging_dir (:obj:`str`, `optional`):
-            `TensorBoard <https://www.tensorflow.org/tensorboard>`_ log directory. Will default to 
+            `TensorBoard <https://www.tensorflow.org/tensorboard>`_ log directory. Will default to
```
Comment (sgugger) [inline src/transformers/training_args_tf.py:100]:
```suggestion
            `TensorBoard <https://www.tensorflow.org/tensorboard>`__ log directory. Will default to
```

**Review by sgugger — APPROVED** [review]
Thanks a lot, this is looking great to me. Just have a few more corrections suggested.

**Comment by lewtun** [comment]
Thanks for the suggestions! I've included them so should be good to go :)

**Comment by sgugger** [comment]
Thanks a lot!

**Pull request merged**